web/libs/ui/src/lib/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,59 @@
+/* eslint-disable-next-line */
+import React, { type InputHTMLAttributes } from "react";
+import styles from "./checkbox.module.scss";
+import { nanoid } from "nanoid";
+
+export interface CheckboxProps extends Omit<InputHTMLAttributes<HTMLInputElement>, "type"> {
+  indeterminate?: boolean;
+}
+const classNameMap = { ...styles };
+
+export const Checkbox = ({ checked, indeterminate, style, onChange, children, ...props }: CheckboxProps) => {
+  const checkboxRef = React.createRef();
+  const withLabel = !!children;
+  const [id] = React.useState(nanoid);
+
+  React.useEffect(() => {
+    if (checkboxRef.current) {
+      checkboxRef.current.indeterminate = indeterminate;
+    }
+  }, [checkboxRef, indeterminate]);
+
+  const checkboxContent = (
+    <span className={styles.checkbox__box}>
+      <input
+        {...props}
+        id={id}
+        ref={checkboxRef as React.Ref<HTMLInputElement>}
+        checked={!!checked}
+        className={styles.checkbox__input}
+        type="checkbox"
+        onChange={(e) => {
+          onChange?.(e);
+        }}
```

Comment:
can we simplify it?
```suggestion
        onChange={onChange}
```
or better just remove it from here and from args spread, so it will just be passed in in `...props` and if it's missing it just won't be set at all